src/script/queue.rs

```diff
@@ -0,0 +1,81 @@
+#[derive(Debug)]
```

Comment:
Decided against it since it was badly documented what happens when you push to it when it's full. It's documented as a ring buffer, so I figured it would override the oldest elements in the queue which I do not want.